python/mxnet/rnn/rnn_cell.py

```diff
@@ -117,21 +118,30 @@ def reset(self):
         self._counter = -1
 
     def __call__(self, inputs, states):
-        """Construct symbol for one step of RNN.
+        """Unroll the RNN for one time step.
 
         Parameters
         ----------
         inputs : sym.Variable
             input symbol, 2D, batch * num_units
-        states : sym.Variable
-            state from previous step or begin_state().
+        states : list of sym.Variable
+            RNN state from previous step or the output of begin_state().
 
         Returns
         -------
         output : Symbol
-            output symbol
-        states : Symbol
-            state to next step of RNN.
+            Symbol corresponding to the output from the RNN when unrolling
+            for a single time step.
+        states : nested list of Symbol
```

Comment:
Good point, I just took the type from the begin_state() function and used that everywhere for consistency. But, I think we can make it list of Symbol everywhere including begin_state() itself.